### DISCUSSION TITLE
Dropdown custom (async) filter

### DISCUSSION THREAD

**Discussion by LoaderB0T** [primefaces/community#538] [PrimeNG]
Hi yall,

First of all, great work with this control library!

Sadly we are missing a key feature in the dropdown component.
We would like to have a way to use a custom asynchronous filter functionality.
The use case is server-side filtering. When the user filters we want to listen to the event and do our web request and then update the options based on that request. This is already possible today, but there is not way to prevent the default filter behavior as far as I can tell.

There would be more than one solution for our use case. Two possible solutions that seen reasonable to implement:

1. Extend `filterMatchMode` by a `noop` or similar filter mode that simply does nothing at all and the developer can implement their own solution.
2. Provide a "cancel" or "preventDefault" functionality in the `DropdownFilterEvent` type of the `(onFilter)` event.

Would this be possible? I would be more than happy to create a PR for either option :)

Thanks & best regards,
Janik

**Comment by cetincakiroglu** [comment]
Hi,

Thanks a lot for the feedback, we're thinking about to except callbacks for filter as we do in autocomplete. Could you please create the PR and tag me so we can review it?

**Reply by LoaderB0T** [reply]
Thanks for getting back to me.
I had a quick look at the autocomplete code and did not immediately find the callback you referred to.
But we could add a filterMatchMode of "callback" or something so that the user can change the datasource based on the query text on their own.
